src/context/chartContext.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { CSSProperties, ReactElement, ReactNode, SVGProps, createContext, useState } from 'react';
+import { AnimationDuration, AnimationTiming, Coordinate, TooltipEventType } from '../util/types';
+import { Global } from '../util/Global';
+import { Payload } from '../component/DefaultTooltipContent';
+
+export type TooltipContentType = ReactElement | ((props: any) => ReactNode);
+
+export type AllowInDimension = {
+  x: boolean;
+  y: boolean;
+};
+
+export type ChartContextType = {
```

Comment:
The context might grow quite a bit, we might consider splitting to one context per component.